Input/ClickHandler.gd

```diff
@@ -49,5 +61,25 @@ func _on_mouse_entered() -> void:
 func _on_mouse_exited() -> void:
 	if not _is_interactable:
 		return
-		
+	
+	_set_lock_hover_timer()
+	
 	on_unhover.emit()
+
+
+func _create_lock_hover_timer():
+	_lock_hover_timer = Timer.new()
+	add_child(_lock_hover_timer)
+	
+	# When timer expires, enable hovering
+	_lock_hover_timer.timeout.connect(func(): _hover_enabled = true)
+
+
+func _set_lock_hover_timer():
+	if _lock_hover_timer == null:
+		return
+	
+	# Disable hovering, set timer. When timer expires, hovering will re-enable.
+	_hover_enabled = false
+	_lock_hover_timer.stop()
```

Comment:
is this used as a way to reset the timer ?

Reply:
Yes, if we entered and the timer was already running, then this will reset it